loop_closure/src/IcpLoopComputation.cc

```diff
@@ -198,6 +202,43 @@ bool IcpLoopComputation::SetupICP(pcl::MultithreadedGeneralizedIterativeClosestP
   return true;
 }
 
+bool IcpLoopComputation::CheckReclosingDistance(gtsam::Key key_from,
+                                                gtsam::Key key_to) const {
+
+  if (closed_keyes_.size() == 0) {
+    return true;
+  }
+
+  // Keyed scans to close to other loop closures
+  gtsam::Key closest_from, closest_to;
+
+  const auto fmit = closed_keyes_.lower_bound(key_from);
+  if (fmit == closed_keyes_.begin())
+    closest_from = *fmit;
+  const auto prev_fmit = std::prev(fmit);
+  closest_from =
+      (fmit == closed_keyes_.end() || key_from - *prev_fmit <= *fmit - key_from)
+          ? *prev_fmit
+          : *fmit;
+
+  const auto toit = closed_keyes_.lower_bound(key_to);
+  if (toit == closed_keyes_.begin())
+    closest_to = *toit;
+  const auto prev_toit = std::prev(toit);
+  closest_to =
+      (toit == closed_keyes_.end() || key_to - *prev_toit <= *toit - key_to)
+          ? *prev_toit
+          : *toit;
+
+  if (abs(static_cast<int>(closest_from) - static_cast<int>(key_from)) <
```

Comment:
does this assume the keys are from the same robot?

Reply:
shouldn't matter, we are using gtsam key here (i.e. it concatenates robot id and pose id ). Essentially allows users to spread out the loop closures to speed up optimization (sparser graph)